### ISSUE TITLE
How to suppress CA1305 warnings for a specific type?

### ISSUE THREAD

**Issue by MartyIX** [dotnet/roslyn-analyzers#5372]
### Analyzer

**Diagnostic ID**: [CA1305](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `Specify IFormatProvider`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.400](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

I have tried to suppress CA1305 for only `NLog.Logger`:

```
dotnet_diagnostic.CA1305.severity = warning

# Not sure if one should use this ...
dotnet_diagnostic.CA1305.excluded_symbol_names = T:NLog.Logger
dotnet_diagnostic.CA1305.excluded_type_names_with_derived_types = T:NLog.Logger

# ... or this
dotnet_code_quality.CA1305.excluded_symbol_names = T:NLog.Logger
dotnet_code_quality.CA1305.excluded_type_names_with_derived_types = T:NLog.Logger
```

and it does not work for me. https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1305 does not mention `excluded_symbol_names` but https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1304 does. Is it unsupported then?

### Expected behavior

`NLog.Logger.Debug(...);` should not report any CA1305 warnings.

### Actual behavior

Each `NLog.Logger.Debug(...);` line emits a CA1305 warning.

## Additional context

This issue is not only about `NLog` package. This is about suppressing the CA1305 warnings for an arbitrary type.

**Comment by Youssef1313** [comment]
I think it's not supported in CA1305. In case you want to contribute a fix, I think this may be a simple `IsConfiguredToSkipAnalysis` call to support that in the following condition:

https://github.com/dotnet/roslyn-analyzers/blob/d0663cca11722045b9aeabb5d0d5335042084e13/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs#L146-L153

Example:

https://github.com/dotnet/roslyn-analyzers/blob/d0663cca11722045b9aeabb5d0d5335042084e13/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyCultureInfo.cs#L54-L57

I think we should somehow issue a warning when an unsupported `dotnet_code_quality` is specified. But not sure if that would be easy/achievable.

**Comment by Evangelink** [comment]
Youssef is right, that's currently not supported but the fix is easy to implement. In case you want to take care of this improvement, please make sure to add a test.


> I think we should somehow issue a warning when an unsupported `dotnet_code_quality` is specified. But not sure if that would be easy/achievable.

That's something I wanted to tackle when I had a bit more time/availability (see #3267). Back then I had some idea on how to do so but I haven't documented it and it slipped out of my mind so I will have to look it up.
